### PR TITLE
Add support for ADR on US_902_928 band

### PR DESCRIPTION
**Pull request by mgranberry**
This pull request adds support for ADR on the US 902-928 band.  The power limits are capped at 20dBm, although the 500kHz channels appear to allow 26dBm.

Related to https://github.com/TheThingsNetwork/ttn/issues/619

### CODE REVIEW

File: core/networkserver/adr.go [hunk 1]
```diff
@@ -168,36 +169,102 @@ func (n *networkServer) handleDownlinkADR(message *pb_broker.DownlinkMessage, de
...
+			sort.Ints(channels)
+
+			chMaskCntl := -1
+			for _, c := range channels {
+				// use the ChMask of the first LinkADRReqPayload, besides
+				// turning off all 125 kHz this payload contains the ChMask
+				// for the last block of channels.
+				if c >= 64 {
+					payloads[0].ChMask[c%16] = true
+					println("Ch", c, true)
```
Comment (htdvisser) [inline core/networkserver/adr.go]: Don't `println`

File: core/networkserver/adr.go [hunk 2]
```diff
@@ -168,36 +169,102 @@ func (n *networkServer) handleDownlinkADR(message *pb_broker.DownlinkMessage, de
...
+							ChMaskCntl: uint8(chMaskCntl),
+							NbRep: uint8(dev.ADR.NbTrans),
+						},
+					}
+
+					// set the channel mask for this block
+					for _, ec := range channels {
+						if ec >= chMaskCntl*16 && ec < (chMaskCntl+1)*16 {
+							pl.ChMask[ec%16] = true
+							println("Ch", c, true)
```
Comment (htdvisser) [inline core/networkserver/adr.go]: Don't `println`

File: core/band/band.go [hunk 3]
```diff
@@ -90,6 +90,18 @@ func Get(region string) (frequencyPlan FrequencyPlan, err error) {
...
 		frequencyPlan.Band, err = lora.GetConfig(lora.US_902_928, false, lorawan.DwellTime400ms)
+		channelGroups := [](int){1} // Enable 903.9-905.3/200 kHz, 904.6 mHz channels
+		for channel := 0; channel < 72; channel++ {
+			for _, channelGroup := range channelGroups {
+				if (channel < channelGroup * 8 || channel >= (channelGroup + 1) * 8) && channel != channelGroup + 64 {
+					frequencyPlan.DisableUplinkChannel(channel)
+				}
+			}
+		}
+		// Is there a reason MaxDataRate shouldn't be 4?  I suppose if it was, the number of channels occupied would be
```
[the remark this answers is not in the document]
Comment (mgranberry) [reply core/band/band.go]: I dug into this a little more.  It looks like it only works as long as everything is ca. 3dB.  The noise floor increases are ~2.5 dB for each SF level and the EU power steps are 3dB.  The 3.5dB difference between SF7BW125 and SF8BW500 is great enough that I'm happy leaving it out, although it seems like it's probably worth experimenting with including the faster rates.

**Comment by mgranberry** [comment]
@htdvisser This latest change adds support for the 2 dB TX power steps that are used in the US band after I identified an issue with some additional tests.  I believe I have now addressed all of the previous comments.

**Comment by htdvisser** [comment]
@mgranberry: I couldn't push to your branch, so we'll continue with this in [feature/us-adr](https://github.com/TheThingsNetwork/ttn/tree/feature/us-adr)

**Pull request closed without merging**